csrc/scheduler/normalization_utils.cpp

```diff
@@ -739,5 +740,197 @@ ScheduleHeuristic getPersistentHeuristicFor(ReductionType reduction_type) {
+  ComputeAtRootDomainMap root_map;
+  root_map.build(true);
+
+  // Helper function to check the pattern equivalence for a list of TensorViews
+  auto checkPattern = [&](const std::vector<TensorView*>& rtvs) -> bool {
+    for (const auto it : c10::irange(1, rtvs.size())) {
+      if (!registry_utils::checkPatternEquivalence(
+              rtvs[it - 1], rtvs[it], root_map)) {
+        scheduler_debug_utils::canScheduleRejectReason(
+            schedule_heuristic,
+            "Unmapped reduction ",
+            rtvs[it - 1],
+            " and ",
+            rtvs[it]);
+        return false;
+      }
+    }
+    return true;
+  };
+
+  // Check the pattern equivalence for the first set of reduction TensorViews
+  if (!checkPattern(reduction_tvs1)) {
+    return false;
+  }
+
+  // Return if there is no second set of reduction TensorViews
+  if (reduction_tvs2.empty()) {
+    return true;
+  }
+
+  // Check the pattern equivalence for the second set of reduction TensorViews
+  // if provided.
+  if (!checkPattern(reduction_tvs2)) {
+    return false;
+  }
+
+  return true;
+}
+
+// The idnetical compile time check of InnerPersistentKernelScheduler and
+// OuterPersistentKernelScheduler.
+bool compileTimeCheck(Fusion* fusion, ScheduleHeuristic schedule_heuristic) {
+  // common checks for all persistent heuristics
+  if (!normalization_scheduler_utils::checkOpsAndInputs(
+          fusion, schedule_heuristic)) {
+    return false;
+  }
+
+  // check reduction types and pattern
+  auto reduction_tvs = scheduler_utils::getReductionTvs(fusion);
+  if (reduction_tvs.empty()) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedule_heuristic, "no reduction tv");
+    return false;
+  }
+  auto reduction_type =
+      reduction_scheduler_utils::getReductionType(reduction_tvs);
+  if (getPersistentHeuristicFor(reduction_type) != schedule_heuristic) {
+    scheduler_debug_utils::canScheduleRejectReason(
+        schedule_heuristic,
+        "schedule_heuristic doesn't match with reduction type.");
+    return false;
+  }
+  if (!checkReductionPattern(fusion, schedule_heuristic, reduction_tvs)) {
+    return false;
+  }
+
+  if (!ir_utils::getViewOps(fusion).empty()) {
+    ComputeAtMap ca_map(fusion);
+    if (registry_utils::requiresForwardViewReplay(fusion, ca_map)) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          schedule_heuristic, "Fusion requires view being reversible.");
+      return false;
+    }
+
+    // Persistent scheduler simply uses reference_tv as the reference, if
+    // that changes, this needs to be changed.
+    auto reference_tv = reduction_tvs[0];
+    if (registry_utils::reductionInterferingView(
+            fusion, ca_map, reference_tv)) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          schedule_heuristic,
+          "View may interfere with normalization scheduling.");
+      return false;
+    }
+  }
+
+  // Before examining the reduction axes want to quickly
```

Comment:
The reference code you provided is from the `ReductionScheduler`, the `canScheduleCompileTime` in `PersistentKernelScheduler` has been changed to this order since the initial commit of this repo.